docs: Add doc/bitcoin-conf.md (#14497)
bitcoin/bitcoin: doc/bitcoin-conf.md

> # `bitcoin.conf` - Bitcoin configuration file

Reviewer: Drop `Bitcoin`?

> All command-line options (except for `-conf`) may be specified in a configuration file, and all configuration file options (except for `-includeconf`) may also be specified on the command line. Command-line options override values set in the configuration file and configuration file options override values set in the GUI.

Reviewer 2: Perhaps mention that certain values need specifying like `testnet=1` in the configuration, but is just `-testnet` on command-line?

> The configuration file is a plain text file and consist of 'setting=value' entries, one per line. Leading and trailing whitespaces are removed. A number sign (`#`) indicates a comment. Comments and empty lines are ignored.

Reviewer 3: Use back-ticks also for `setting=value` to make it consistent?

Reviewer 4: maybe "(`#`) indicates the start a comment, and will make the parser ignore until the end of the line"

> The configuration file is a plain text file and consist of 'setting=value' entries, one per line. Leading and trailing whitespaces are removed. A number sign (`#`) indicates a comment. Comments and empty lines are ignored.

Reviewer: Could mention that, although possible, it's not recommended to have a setting and a comment in the same line?
```
foo=bar # reasons
```
vs
```
# reasons
foo=bar
```

> The configuration file is a plain text file and consist of 'setting=value' entries, one per line. Leading and trailing whitespaces are removed. A number sign (`#`) indicates a comment. Comments and empty lines are ignored.

Reviewer 5: ```suggestion
The configuration file is a plain text file and consists of 'setting=value' entries, one per line. Leading and trailing whitespaces are removed. A number sign (`#`) indicates a comment. Comments and empty lines are ignored.
```

The configuration file is not automatically created; you can create it using your favorite plain-text editor. By default, the configuration file name is `bitcoin.conf` and it is located in the Bitcoin data directory, but both the Bitcoin data directory and the configuration file path may be changed using the `-datadir` and `-conf` command-line arguments.

The `includeconf=<file>` option in the `bitcoin.conf` file can be used to include additional configuration files.

> Network specific options can be placed into the sections with headers `[main]` (not `[mainnet]`), `[test]` (not `[testnet]`) and `[regtest]`.

Reviewer 6: I believe you can also write network specific options by prefixing the option with `netname.`.

Reviewer 4: yes, though I think in general when writing user documentation, documenting one way that works (and is recommended) is good enough; the conversion to `a.b` might be seen as an implementation detail that will just confuse

but don't care strongly, could add it as extra information

> Network specific options can be placed into the sections with headers `[main]` (not `[mainnet]`), `[test]` (not `[testnet]`) and `[regtest]`.

Reviewer 5: ```suggestion
Network specific options can be placed into sections with headers `[main]` (not `[mainnet]`), `[test]` (not `[testnet]`) or `[regtest]`.
```

## Configuration File Format

The configuration file is a plain text file and consists of `option=value` entries, one per line. Leading and trailing whitespaces are removed.

In contrast to the command-line usage:
- an option must be specified without leading `-`;
- a value of the given option is mandatory; e.g., `testnet=1` (for chain selection options), `noconnect=1` (for negated options).

### Blank lines

Blank lines are allowed and ignored by the parser.

### Comments

A comment starts with a number sign (`#`) and extends to the end of the line. All comments are ignored by the parser.

Comments may appear in two ways:
- on their own on an otherwise empty line (_preferable_);
- after an `option=value` entry.

### Network specific options

Network specific options can be:
- placed into sections with headers `[main]` (not `[mainnet]`), `[test]` (not `[testnet]`) or `[regtest]`;
- prefixed with a chain name; e.g., `regtest.maxmempool=100`.

## Configuration File Path

> The configuration file is not automatically created; you can create it using your favorite plain-text editor. By default, the configuration file name is `bitcoin.conf` and it is located in the Bitcoin data directory, but both the Bitcoin data directory and the configuration file path may be changed using the `-datadir` and `-conf` command-line options.

Reviewer 5: Nit: above you write "plain text" as two words, here you write it hyphenated. Not sure which version is better, but it would probably be good to be consistent
